server/src/lib.rs

```diff
@@ -2050,6 +2054,83 @@ mod tests {
         assert!(server.db_names_sorted().contains(&String::from("foo")));
     }
 
+    #[tokio::test]
+    async fn init_too_many_active_generation_directories() {
+        let application = make_application();
+        let server_id = ServerId::try_from(1).unwrap();
+
+        let server = make_server(Arc::clone(&application));
+        server.set_id(server_id).unwrap();
+        server.wait_for_init().await.unwrap();
+
+        let foo_db_name = DatabaseName::new("foo").unwrap();
+
+        // Create database
+        create_simple_database(&server, &foo_db_name)
+            .await
+            .expect("failed to create database");
+
+        // Delete it
+        server.delete_database(&foo_db_name).await.unwrap();
+
+        // Create it again
+        create_simple_database(&server, &foo_db_name)
+            .await
+            .expect("failed to create database");
+
+        // Delete it again
+        server.delete_database(&foo_db_name).await.unwrap();
+
+        std::mem::drop(server);
+
+        // Remove the tombstone files from both database generation directories
+        let mut db_path = application.object_store().new_path();
+        db_path.push_all_dirs(&[server_id.to_string().as_str(), foo_db_name.as_str()]);
+        let database_files: Vec<_> = application
+            .object_store()
+            .list(Some(&db_path))
+            .await
+            .unwrap()
+            .try_collect::<Vec<_>>()
+            .await
+            .unwrap()
+            .into_iter()
+            .flatten()
+            .collect();
+
+        // Delete all tombstone files
+        for file in database_files {
+            let parsed: DirsAndFileName = file.clone().into();
+            if parsed.file_name.unwrap().to_string() == "DELETED" {
+                application.object_store().delete(&file).await.unwrap();
+            }
+        }
```

Comment:
```suggestion
        // Delete all tombstone files
        let mut deleted_something = false;
        for file in database_files {
            let parsed: DirsAndFileName = file.clone().into();
            if parsed.file_name.unwrap().to_string() == "DELETED" {
                application.object_store().delete(&file).await.unwrap();
                deleted_something = true;
            }
        }
        assert!(deleted_something);
```

If the tombstone pattern ever changes, we should ensure that this test still deletes them. I'm using a similar pattern when messing around w/ the catalog. While for sure it's not perfect it's at least some safety net.